### DISCUSSION TITLE
Issues with the mod

### DISCUSSION THREAD

**Discussion by ZetaMLG** [qwbarch/mirage#120] [Q&A]
When playing, the mod doesn't work (along with other mods unrelated to skinwalkers), and I noticed this error in the log:

[03:17:20.7790534] [Error  : Unity Log] TypeLoadException: Could not resolve type with token 0100001a from typeref (expected class 'Mirage.Core.Task.Channel' in assembly 'Mirage.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null')
Stack trace:
System.Runtime.CompilerServices.AsyncTaskMethodBuilder1[TResult].Start[TStateMachine] (TStateMachine& stateMachine) (at <787acc3c9a4c471ba7d971300105af24>:IL_002C)
<StartupCode$Mirage>.$Plugin.func1@1-7 (System.Threading.CancellationToken cancellationToken, Microsoft.FSharp.Core.FSharpFunc2[T,TResult] program) (at <a58e13ec8833f0e0b5e7b86206380bd8>:IL_0029)
Mirage.Plugin.Awake () (at ./Mirage/Plugin.fs:40)
UnityEngine.GameObject:DMD<UnityEngine.GameObject::AddComponent>(GameObject, Type)
BepInEx.Bootstrap.Chainloader:DMD<BepInEx.Bootstrap.Chainloader::Start>()
UnityEngine.Rendering.HighDefinition.HDRenderPipelineAsset:OnEnable()

When playing, the mod doesn't detect the microphone, causing it not to work (I have the mod and its dependencies updated).

**Comment by qwbarch** [comment]
This looks like an installation issue. Are you using a mod manager? If you do have a mod manager installed, how long is your profile name?

A very long profile name has been known to cause issues with Window's 255 character limit in file paths, which can cause some dlls to not load.

**Reply by ZetaMLG** [reply]
My username is no longer than 9 characters, I use Thunderstore as my mod manager and the mod path does not exceed 136 characters.

**Reply by qwbarch** [reply]
That's odd then. Can you send me a full log, as well as your profile code?

**Reply by ZetaMLG** [reply]
Hi, sorry for the delay. Let me tell you what happened:

I was testing, and the problem was mainly that the mod was having issues with Thunderstore. Even though I applied the update, it wouldn’t execute the mod correctly. It was strange because I went into Mirage's settings, and everything seemed to be configured. However, after several attempts, the configuration reset itself. That’s when it started working, clearing the compiled error in the log. I’m not sure about the exact cause, but for now, it seems to be working again.

PS: It would have been interesting if a hallucination mechanic had been implemented during Mirage’s development. For example, besides the Masked, the player could also see other players that don’t exist. This would be similar to what the InsanityRemastered mod used to do (Insanity stopped working a long time ago).

**Reply by qwbarch** [reply]
Glad to see it's working for you!

As for feature requests, Mirage is feature complete and I don't plan on adding anything else anymore. I'm only going to be maintaining the mod to make sure it works with Zeeker's game updates, as well as bug fixes, but other than that I won't be working on it anymore
